modules/git/repo_commit.go

```diff
@@ -227,11 +222,38 @@ func (repo *Repository) searchCommits(id SHA1, opts SearchCommitsOptions) (*list
 	if opts.All {
 		cmd.AddArguments("--all")
 	}
-	stdout, err := cmd.RunInDirBytes(repo.Path)
+	var stdout []byte
+	if len(opts.Keywords) > 0 {
+		initialStdout, err := cmd.RunInDirBytes(repo.Path)
```

Comment:
I don't think this is right, because there is a prameter `-100` before.

Reply:
I didn't think about that. The commit hash search would not go back as far as the grep search.